### ISSUE TITLE
twisted.conch.client.knownhosts has many coding standard violations

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6257]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6257|
|Type|defect|
|Created|2013-01-15 00:01:22Z|
|Branch|https://github.com/twisted/twisted/tree/knownhosts-standards-6257|

As reported by `twistedchecker`.  Most -- or, who knows, perhaps even all -- of these should be fixed.

<details><summary>Searchable metadata</summary>

```
trac-id__6257 6257
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_knownhosts_standards_6257 branches/knownhosts-standards-6257
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1358208082000000 1358208082000000
changetime__1359405245000000 1359405245000000
version__None None
owner__exarkun exarkun
cc__z3p cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [36815]) Branching to 'knownhosts-standards-6257'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

I fixed many, but not all.  I think some of the remaining errors may not make sense.  For convenience, here they are as reported by a twistedchecker build of the branch:

```
************* Module twisted.conch.client.knownhosts
W9207: 75,0:_BaseEntry: Missing a blank line before epytext markups
W9207:111,0:PlainEntry: Missing a blank line before epytext markups
W9202:206,4:UnparsedEntry.matchesHost: Missing epytext markup @param for argument "hostname"
W9204:206,4:UnparsedEntry.matchesHost: Missing epytext markup @return for return value
W9202:213,4:UnparsedEntry.matchesKey: Missing epytext markup @param for argument "key"
W9204:213,4:UnparsedEntry.matchesKey: Missing epytext markup @return for return value
W9208:419,16:KnownHostsFile.verifyHostKey.gotHasKey.promptResponse: Missing docstring
```

[Complete build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/knownhosts-standards-6257)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

1.`@rtype` is missing for `_extractCommon`
 1. `@param opener` in `ConsoleUI` is missing a `@type opener`
 1. the empty line on 405 isn't needed
 1. `@type` should be `@rtype` on line 452

Please merge after that, thanks.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [36854]) Address most review feedback

refs [#6257](https://github.com/twisted/twisted/issues/6257)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

> @param opener in ConsoleUI is missing a @type opener

There's neither a type nor an interface that could be put there.  Do you have a suggestion?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_8"></a>|@glyph set owner to @exarkun|
|-|-|

Replying to [exarkun](#note_6):
> > @param opener in ConsoleUI is missing a @type opener
> 
> There's neither a type nor an interface that could be put there.  Do you have a suggestion?

"`callable taking no arguments, returning L{None}`"

Other than that this looks fine to me; please land.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [36996]) some stuff for the @type of opener

refs [#6257](https://github.com/twisted/twisted/issues/6257)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `closed`|
|-|-|

(In [36997]) Merge knownhosts-standards-6257

Author: exarkun
Reviewer: thijs, glyph
Fixes: [#6257](https://github.com/twisted/twisted/issues/6257)

Improve `twisted.conch.client.knownhosts` compliance with API documentation
standards.
